test/unit/WrapperRegistry.ts

```diff
@@ -0,0 +1,303 @@
+import { ethers } from 'hardhat'
+import { BigNumber, Contract, Signer } from 'ethers'
+import { expect } from 'chai'
+
+interface TestContext {
+  accounts: Signer[]
+  deployer: Signer
+  userA: Signer
+  wrapperRegistry: Contract
+  underlyingTokenAddressA: string
+  underlyingTokenAddressB: string
+  wrapperTokenAddressA: string
+  wrapperTokenAddressB: string
+}
+
+describe('WrapperRegistry', () => {
+  const setupTestContext = async (): Promise<TestContext> => {
+    const accounts: Signer[] = await ethers.getSigners()
+    const [deployer, userA] = accounts
+
+    const wrapperRegistryFactory = await ethers.getContractFactory(
+      'WrapperRegistry',
+    )
+    const wrapperRegistry = await wrapperRegistryFactory
+      .connect(deployer)
+      .deploy()
+
+    const underlyingTokenAddressA: string =
+      '0x000000000000000000000000000000000000000A'
+    const underlyingTokenAddressB: string =
+      '0x000000000000000000000000000000000000000b'
+    const wrapperTokenAddressA: string =
+      '0x0000000000000000000000000000000000000001'
+    const wrapperTokenAddressB: string =
+      '0x0000000000000000000000000000000000000002'
+
+    return {
+      accounts,
+      deployer,
+      userA,
+      wrapperRegistry,
+      underlyingTokenAddressA,
+      underlyingTokenAddressB,
+      wrapperTokenAddressA,
+      wrapperTokenAddressB,
+    }
+  }
+
+  describe('Initialization', function () {
+    it('Can successfully deploy BondConfigVault with proper arguments', async () => {
+      const { wrapperRegistry } = await setupTestContext()
+      expect(await wrapperRegistry.numWrappers()).to.eq(0)
+      await expect(wrapperRegistry.wrapperAt(0)).to.be.reverted
+    })
+  })
+
+  describe('Ownership', function () {
+    it('Can transfer ownership', async () => {
+      const { wrapperRegistry, deployer, userA } = await setupTestContext()
+      const deployerAddress = await deployer.getAddress()
+      const userAAddress = await userA.getAddress()
+
+      expect(await wrapperRegistry.owner()).to.eq(deployerAddress)
+      await wrapperRegistry.transferOwnership(userAAddress)
+      expect(await wrapperRegistry.owner()).to.eq(userAAddress)
+    })
+  })
+
+  describe('Simple Updating Wrappers', function () {
+    it('Can successfully add a wrapper', async () => {
+      const {
+        wrapperRegistry,
+        underlyingTokenAddressA,
+        wrapperTokenAddressA,
+      } = await setupTestContext()
+
+      await expect(
```

Comment:
You should test the return values too. (ie returns true/false as per your definition)

You can do this by using `callStatic`
```
const r = await wrapperRegistry.callStatic.addWrapper(
  underlyingTokenAddressA,
  wrapperTokenAddressA,
)
expect(r).to.be.true
```